kubernetes.go

```diff
@@ -0,0 +1,682 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	secretPKI        = "openvpn-pki-ca"
+	secretServer     = "openvpn-pki-server"
+	secretClientTmpl = "openvpn-pki-%s"
+	secretCRL        = "openvpn-pki-crl"
+	secretIndexTxt   = "openvpn-pki-index-txt"
+	secretDHandTA    = "openvpn-pki-dh-and-ta"
+	certFileName     = "pem.crt"
+	privKeyFileName  = "pem.key"
```

Comment:
Can't we use Kubernetes secrets `type: kubernetes.io/tls`?
https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets

It is known and it widely used everywhere

```suggestion
	certFileName     = "tls.crt"
	privKeyFileName  = "tls.key"
	caCertFileName   = "ca.crt"
```

You can also inject `ca.crt` into each secret as it doing cert-manager